develop/app/Controllers/v2/ProductController.php

```diff
@@ -0,0 +1,299 @@
+            "list"   => [],
+            "dataCount" => $dataCount
+        ];
+
+        if ($product) {
+            foreach ($product as $productEntity) {
+                $productData = [
+                    "p_key"       => $productEntity->p_key,
+                    "name"        => $productEntity->name,
+                    "price"       => $productEntity->price,
+                    "amount"      => $productEntity->amount,
+                    "createdAt"   => $productEntity->createdAt,
+                    "updatedAt"   => $productEntity->updatedAt
+                ];
+                $data["list"][] = $productData;
+            }
+        } else {
+            return $this->fail("product data not found", 404);
+        }
+
+
+        return $this->respond([
+            "status" => true,
+            "data"  => $data,
+            "msg" => "product index method successful."
+        ]);
+    }
+
+
+    /**
+     * [GET] api/v2/product/{p_key}
+     * get someone product by p_key
+     *
+     * @param integer $p_key
+     * @return void
+     */
+    public function show($p_key = null)
+    {
+        if (is_null($p_key)) {
+            return $this->fail("Incoming data(Product Key) not true", 400);
+        }
+
+        $productEntity = new ProductEntity();
+        $productModel  = new ProductModel();
+
+        $productEntity = $productModel->find($p_key);
+
+        if ($productEntity) {
+            $data = [
+                "p_key"       => $productEntity->p_key,
+                "name"        => $productEntity->name,
+                "price"       => $productEntity->price,
+                "amount"       => $productEntity->amount,
+                "createdAt"   => $productEntity->createdAt,
+                "updatedAt"   => $productEntity->updatedAt
+            ];
+        } else {
+            return $this->fail("product data not found", 404);
+        }
+
+        return $this->respond([
+            "data" => $data,
+            "msg" => "product show method successful."
+        ]);
+    }
+
+    /**
+     * [POST] api/v2/product/
+     * create product
+     *
+     * @return void
+     */
+    public function create()
+    {
+        $data   = $this->request->getJSON(true);
+
+        $name   = $data["name"]   ?? null;
+        $price  = $data["price"]  ?? null;
+        $amount = $data["amount"] ?? null;
+
+        if (is_null($name) || is_null($price) || is_null($amount)) {
+            return $this->fail("Incoming data error", 400);
+        }
+
+        $productModel = new ProductModel();
+        $productCreateResult =  $productModel->createProductTransaction($name, $price, $amount);
+
+        if ($productCreateResult) {
+            return $this->respond([
+                "status" => true,
+                "data" => $productCreateResult,
+                "msg" => "product create method successful."
+            ]);
+        } else {
+            return $this->fail("product create method fail.", 400);
```

Comment:
The error content should be `The product create failed, please try again.`

Reply:
This chahge is done.